src/howitz/endpoints.py

```diff
@@ -477,28 +477,79 @@ def show_update_events_status_modal():
 
 @main.route('/event/<i>/unselect', methods=["GET"])
 def unselect_event(i):
+    event_type = request.args.get('eventtype', '')
     try:
-        session["selected_events"].remove(i)
+        session["selected_events"].pop(str(i), None)
         session.modified = True
         current_app.logger.debug("SELECTED EVENTS %s", session["selected_events"])
     except ValueError:
         pass
 
+    show_clear_flapping = all(event == 'portstate' for event in session.get("selected_events", dict(k='v')).values())
+
     return render_template('/responses/toggle-select.html', id=i, is_checked=False,
-                           is_menu=len(session["selected_events"]) > 0)
+                           is_menu=len(session["selected_events"]) > 0, event_type=event_type,
+                           show_clear_flapping=show_clear_flapping)
 
 
 @main.route('/event/<i>/select', methods=["GET"])
 def select_event(i):
+    event_type = request.args.get('eventtype', '')
     try:
-        session["selected_events"].append(i)
+        session["selected_events"][str(i)] = event_type
```

Comment:
What happens if event_type is the empty string?

Reply:
And "i" is a bit short. Is it an event_id? Call it "event_id". Is it the numerical index in a list? Call it "index".

Also, which line raises the expected ValueError?